### ISSUE TITLE
[BUG] boto3-stubs breaks mypy evaluation of boto3.Session

### ISSUE THREAD

**Issue by corleyma** [vemel/mypy_boto3_builder#79]
**Describe the bug**
After installing boto3-stubs, referencing boto3.Session will cause mypy to throw a spurious "Module has no attribute" error.

**To Reproduce**
1. Install `boto3-stubs` and `boto3`
2.  Run mypy on the following code sample

```python
import boto3

session: boto3.Session = boto3.Session()

```
**Actual mypy output**

```
error: Module has no attribute "Session"
```

**Expected mypy-output**

```
Success: no issues found in 1 source files
```

**Additional context**
MacOS and Debian, `boto3-stubs == 1.17.84` installed via pip.

**Comment by vemel** [comment]
Meanwhile you can use 
```python
from boto3.session import Session
```

WIll be fixed in the next release.

**Comment by vemel** [comment]
I have just tried to add `__all__` explicitly and `mypy` still complains about missing `boto3.Session`. Please import `Session` from `boto3.session`.

I will try to find a solution, but I have no idea how to fix it now.

**Comment by corleyma** [comment]
Yeah, it's worth noting that even trying to do
```python
import boto3

s = boto3.session.Session()
```

will make mypy unhappy if boto3-stubs is installed.

**Comment by corleyma** [comment]
@vemel Does doing something like `from boto3.session import Session as Session` (in `__init__.pyi` specifically) make mypy happy?  Adding to `__all__` _should_ work these days, but this kind of explicit re-export w/ alias (any from-as import) should also work too.

**Comment by vemel** [comment]
Makes sense. Thank you for the idea!

**Comment by vemel** [comment]
I have just released a fix, which should be included in `boto3 1.17.79`. Please let me know if it works for you as expected.

**Comment by corleyma** [comment]
Just tested and this looks good to me, although I think we have a similar issue with botocore (where mypy thinks botocore.session botocore.credentials, etc, don't exist if the stubs are installed).  Should be a similar fix.

**Comment by corleyma** [comment]
Updating: boto3.Session does work as expected now, but we're still missing boto3.session.  So e.g. 

```python3
import boto3

session = boto3.session.Session()
```
will make mypy unhappy.

**Comment by vemel** [comment]
Probably this is because of a conflict between `boto3/__init__.py session.Session` and `boto3/session.py Session`. I will take a look.

**Comment by vemel** [comment]
@corleyma the newest `mypy 0.901` looks happy. However, `pyright` still reports the error. I am actually on `pyright`'s side, there is a name conflict that cannot be resolved properly.

What do you suggest?
